spring-data-couchbase/src/main/java/org/springframework/data/couchbase/core/convert/CryptoConverter.java

```diff
@@ -0,0 +1,166 @@
+				try {
+					return svc.convert(decryptedString, context.getProperty().getType());
+				} catch (Exception e) {
+					exceptionList.add(e);
+				}
+			}
+			if (conversions.hasCustomReadTarget(Long.class, context.getProperty().getType())) {
+				try {
+					return svc.convert(Long.valueOf(decryptedString), property.getType());
+				} catch (Exception e) {
+					exceptionList.add(e);
+				}
+			}
+			if (conversions.hasCustomReadTarget(Double.class, context.getProperty().getType())) {
+				try {
+					return svc.convert(Double.valueOf(decryptedString), property.getType());
+				} catch (Exception e) {
+					exceptionList.add(e);
+				}
+			}
+			if (conversions.hasCustomReadTarget(Boolean.class, context.getProperty().getType())) {
+				try {
+					Object booleanResult = svc.convert(Boolean.valueOf(decryptedString), property.getType());
+					if (booleanResult == null) {
+						throw new Exception("value " + decryptedString + " would not convert to boolean");
+					}
+				} catch (Exception e) {
+					exceptionList.add(e);
+				}
+			}
+			// let's try to find a constructor...
+			try {
+				Constructor<?> constructor = context.getProperty().getType().getConstructor(String.class);
+				return constructor.newInstance(decryptedString);
+			} catch (NoSuchMethodException | InstantiationException | IllegalAccessException | InvocationTargetException e) {
+				exceptionList.add(new Exception("tried to instantiate from constructor taking string arg but got " + e));
+			}
+			// last chance...
+			try {
+				return ctx.getConverter().getPotentiallyConvertedSimpleRead(decryptedString, property);
+			} catch (Exception e) {
+				exceptionList.add(e);
+				RuntimeException ee = new RuntimeException(
+						"failed to convert " + decryptedString + " due to the following suppressed reasons(s): ");
+				exceptionList.stream().forEach(ee::addSuppressed);
+				throw ee;
+			}
+
+		} else {
+			CouchbaseDocument decryptedDoc = new CouchbaseDocument().setContent(JsonObject.fromJson(decrypted));
+			CouchbasePersistentEntity<?> entity = ctx.getConverter().getMappingContext()
+					.getRequiredPersistentEntity(property.getType());
+			return ctx.getConverter().read(entity, decryptedDoc, null);
+		}
+	}
+
+	@Override
+	public CouchbaseDocument write(Object value, ValueConversionContext<? extends PersistentProperty<?>> context) {
+		byte[] plainText;
+		CouchbaseConversionContext ctx = (CouchbaseConversionContext) context;
+		CouchbasePersistentProperty property = ctx.getProperty();
+		org.springframework.data.convert.CustomConversions conversions = ctx.getConverter().getConversions();
+
+		Class<?> sourceType = context.getProperty().getType();
+		Class<?> targetType = conversions.getCustomWriteTarget(context.getProperty().getType()).orElse(null);
+
+		value = ctx.getConverter().getPotentiallyConvertedSimpleWrite(property, ctx.getAccessor(), false);
+		if (conversions.isSimpleType(sourceType)) {
```

Comment:
Similar comment to above - I'm not really happy with this code that converts the value into a string for decryptiing.  But it seems to work.